larky/src/main/java/com/verygood/security/larky/modules/jks/JKS.java

```diff
@@ -0,0 +1,96 @@
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JKS implements StarlarkValue {
+
+    public static final JKS INSTANCE = new JKS();
+
+    @StarlarkMethod(
+            name = "load_key_and_certificates",
+            parameters = {
+                    @Param(name = "keystore_bytes", allowedTypes = {
+                            @ParamType(type = StarlarkBytes.class)
+                    }),
+                    @Param(name = "keystore_password", allowedTypes = {
+                            @ParamType(type = StarlarkBytes.class),
+                            @ParamType(type = NoneType.class),
+                    }),
+                    @Param(name = "key_alias", allowedTypes = {
+                            @ParamType(type = StarlarkBytes.class),
+                    }),
+                    @Param(name = "key_password", allowedTypes = {
+                            @ParamType(type = StarlarkBytes.class),
+                    }),
+            },
+            useStarlarkThread = true
+    )
+    public Tuple loadKeyAndCertificates(
+            final StarlarkBytes larkyKeystoreBytes,
+            final Object larkyKeystorePassword,
+            final StarlarkBytes larkyKeyAlias,
+            final StarlarkBytes larkyKeyPassword,
+            final StarlarkThread thread
+    ) throws EvalException {
+
+        byte[] keystoreBytes = larkyKeystoreBytes.toByteArray();
+        final char[] keystorePassword = Starlark.isNullOrNone(larkyKeystorePassword)
+                ? null
+                : ((StarlarkBytes) larkyKeystorePassword).decode("utf-8", "report").toCharArray();
+        String keyAlias = larkyKeyAlias.decode("utf-8", "report");
+        char[] keyPassword = larkyKeyPassword.toCharArray();
+
+        LarkyKeyPair keyPair;
+        List<LarkyX509Certificate> certificateChain = new ArrayList<>();
+
+        try (InputStream is = new ByteArrayInputStream(keystoreBytes)) {
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keyStore.load(is, keystorePassword);
+            PrivateKey privateKey = (PrivateKey) keyStore.getKey(keyAlias, keyPassword);
+            PublicKey publicKey = keyStore.getCertificate(keyAlias).getPublicKey();
+            keyPair = new LarkyKeyPair(publicKey, privateKey);
+            for (Certificate certificate : keyStore.getCertificateChain(keyAlias)) {
+                LarkyX509Certificate larkyCertificate = LarkyX509Certificate.of((X509Certificate) certificate);
+                certificateChain.add(larkyCertificate);
+            }
+        } catch (KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException | IOException |
+                 CertificateException e) {
+            throw new EvalException(e);
+        }
+
+        return Tuple.of(
+                keyPair,
+                certificateChain.remove(0),
```

Comment:
a bit unclear what's happening here

Reply:
Basically we emulate behaviour from Python library, I think putting a link to the Python code should be good

Reply:
This is actually an analogy of this method but with JKS under the hood
https://github.com/verygoodsecurity/starlarky/blob/master/larky/src/main/java/com/verygood/security/larky/modules/openssl/OpenSSL.java#L242-L257